contracts/Organization.sol

```diff
@@ -0,0 +1,272 @@
+     */
+    function initiateOwnershipTransfer(address _proposedOwner)
+        external
+        onlyOwner
+        returns (bool)
+    {
+        require(
+            _proposedOwner != owner,
+            "ProposedOwner address can't be current owner address."
+        );
+
+        proposedOwner = _proposedOwner;
+
+        emit OwnershipTransferInitiated(_proposedOwner);
+
+        return true;
+    }
+
+    /**
+     * @notice Complete ownership transfer to proposed owner.
+     *
+     * @dev Requires:
+     *          - msg.sender should be proposed owner.
+     *
+     * @return true on successful execution.
+     */
+    function completeOwnershipTransfer() external returns (bool)
+    {
+        require(
+            msg.sender == proposedOwner,
+            "Caller is not proposed owner address."
+        );
+
+        address oldOwner = owner;
+        owner = proposedOwner;
+        proposedOwner = address(0);
+
+        emit OwnershipTransferCompleted(oldOwner, owner);
+
+        return true;
+    }
+
+    /**
+     * @notice Sets admin address.
+     *
+     * @dev Requires:
+     *          - msg.sender should be owner or admin.
+     *          - admin should not be same as owner.
+     *      Allows resetting of admin address to 0x.
+     *
+     * @param _admin admin address to be added.
+     *
+     * @return true on successful execution.
+     */
+    function setAdmin(address _admin) external onlyOrganization returns (bool)
+    {
+        require(
+            _admin != owner,
+            "Admin address can't be owner address."
+        );
+
+        admin = _admin;
+
+        emit AdminAddressChanged(admin);
+
+        return true;
+    }
+
+    /**
+     * @notice Sets worker and its expiration height.
+     *
+     * @dev Requires:
+     *          - Caller should be owner or admin.
+     *          - worker address can't be null.
+     *          - expiration height should not be expired.
+     *          - 0 expiration height is not allowed.
+     *      Admin/Owner has the flexibility to extend/reduce worker expiration
+     *      height. This way a worker activation/deactivation can be
+     *      controlled without adding/removing new worker keys.
+     *
+     * @param _worker worker address to be added.
+     * @param _expirationHeight expiration height of worker.
+     *
+     * @return Remaining height for which worker is active.
+     */
+    function setWorker(
+        address _worker,
+        uint256 _expirationHeight
+    )
+        external
+        onlyOrganization
+        returns (uint256 remainingHeight_)
+    {
+        require(
+            _worker != address(0),
+            "Worker address is null."
+        );
+
+        // expiration height with 0 block number is not allowed since the
+        // minimum block number can be 0.
+        require(
+            _expirationHeight > block.number,
+            "Expiration height is lte to the current block height."
```

Comment:
While this statement is technically correct, because block height and block number are not by definition the same, I would suggest referring to block number to minimize any confusion.